src/box/memtx_tx.h

```diff
@@ -151,6 +150,13 @@ struct memtx_story {
 	 * Number of indexes in this space - and the count of link[].
 	 */
 	uint32_t index_count;
+	/** Status of story for memory manager. */
+	int pin_status;
+	/**
+	 * Flag is set when @a tuple is not placed in primary key and
+	 * the story is the only reason why @a tuple cannot be deleted.
+	 */
+	bool is_pinned;
```

Comment:
Could we use pin_status instead of additional bool?

Reply:
AFAICS, sometimes a story with tuple placed in space and referenced by primary key cannot be deleted because of potential read view or direct usage, so story status and flag is_pinned are independent and the flag cannot be replaced with a special status.